### PR TITLE
[DOC] Add docs how to enable monitoring and alerting of the operator

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
**What this PR does**:
Add docs how to enable monitoring and alerting of the operator, and clarify the requirement of the Prometheus Operator (or a comparable solution).

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/setup/operator/monitor.md [hunk 1]
```diff
@@ -87,3 +92,21 @@ spec:
       sampling_fraction: "1.0"
       jaeger_agent_endpoint: localhost:6831
 ```
+
+
+## Monitor the operator
+
+The Tempo Operator can expose upgrade and other operational metrics about the operator itself, and can create alerts based on these metrics.
+The configuration options to enable the creation of `ServiceMonitor` (for scraping metrics) and `PrometheusRule` (for creating alerts) objects is in the ConfigMap `tempo-operator-manager-config` in the same namespace as the operator:
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/monitor.md:100]: This second sentence is a little confusing. Do you mean that someone has to add the example config map that you have below or that they need to set createServiceMonitors and createPrometheusRules to true in their ConfigMap?

File: docs/sources/tempo/setup/operator/monitor.md [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+title: Monitor Tempo instances and the operator
+description: Set up monitoring for Tempo instances and the operator
 menuTitle: Monitor
 weight: 300
 aliases:
 - /docs/tempo/operator/monitor
 ---
 
-# Monitor TempoStack instances
+# Monitor Tempo instances and the operator
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/monitor.md:10]: Good title update! Thank you for updating the title in front matter as well.

Comment (knylander-grafana) [reply docs/sources/tempo/setup/operator/monitor.md:10]: Please feel free to edit that suggestion.

File: docs/sources/tempo/setup/operator/monitor.md [hunk 3]
```diff
@@ -87,3 +92,21 @@ spec:
       sampling_fraction: "1.0"
       jaeger_agent_endpoint: localhost:6831
 ```
+
+
+## Monitor the operator
+
+The Tempo Operator can expose upgrade and other operational metrics about the operator itself, and can create alerts based on these metrics.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/monitor.md:99]: What do you mean by "can expose upgrade and other operational metrics"?

File: docs/sources/tempo/setup/operator/monitor.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/tempo/setup/operator/monitor.md:100]: Is this correct? 
```suggestion
To enable metrics, you need to create the `ServiceMonitor` (for scraping metrics) and `PrometheusRule` (for creating alerts) objects. These objects are configured in the ConfigMap `tempo-operator-manager-config` in the same namespace as the operator. 

To configure these objects, set `createServiceMonitors` and `crreatePrometheusRules` to `true`. 
```

**Review by knylander-grafana — CHANGES_REQUESTED** [review]
Good updates to the doc. I have some questions about the content.

File: docs/sources/tempo/setup/operator/monitor.md (hunk 1 above)
Comment (andreasgerstmayr) [reply docs/sources/tempo/setup/operator/monitor.md]: This config map already exists, it's part of the installation of the operator. This configmap contains various settings for the operator.

When the `createServiceMonitors` setting is enabled, the operator creates this `ServiceMonitor` object in the cluster, and another operator in the cluster (for example Prometheus Operator) reads this object and configures monitoring. The user does not create this `ServiceMonitor` object, this part is handled by the operator, if the `createServiceMonitors` setting is enabled.

I've rephrased this in my last commit, I hope this clears things up? If not, please let me know :)

File: docs/sources/tempo/setup/operator/monitor.md (hunk 3 above)
Comment (andreasgerstmayr) [reply docs/sources/tempo/setup/operator/monitor.md:102]: The operator handles Tempo upgrades, and it exposes metrics like "number of successful Tempo upgrades", "number of failed Tempo upgrades". And the operator create alerts, to notify system administrators if for example any Tempo upgrade failed.

The other operational metrics are internals of the operator, for example the duration of a "reconcile loop iteration". That's a operator-specific component which continuously tries to match the expected state as described in the TempoStack custom resource to the actual cluster state. E.g. if you delete some object in the cluster which is managed by the operator, the operator re-creates this object again, to match the expected state of the cluster.

File: docs/sources/tempo/setup/operator/monitor.md [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
 menuTitle: Monitor
 weight: 300
 aliases:
 - /docs/tempo/operator/monitor
 ---
 
-# Monitor TempoStack instances
+# Monitor Tempo instances and the operator
+
+You can configure the Tempo Operator to monitor TempoStack instances (including all Tempo components like the distributor). In addition, the operator can expose metrics about the operator itself (for example the number of successful and failed upgrades, etc.).
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/monitor.md:12]: Adding a missing comma.
```suggestion
You can configure the Tempo Operator to monitor TempoStack instances (including all Tempo components like the distributor). In addition, the operator can expose metrics about the operator itself (for example, the number of successful and failed upgrades, etc.).
```

File: docs/sources/tempo/setup/operator/monitor.md (hunk 3 above)
Comment (knylander-grafana) [reply docs/sources/tempo/setup/operator/monitor.md:102]: This is great info!

File: docs/sources/tempo/setup/operator/monitor.md [hunk 5]
```diff
@@ -87,3 +95,23 @@ spec:
       sampling_fraction: "1.0"
       jaeger_agent_endpoint: localhost:6831
 ```
+
+
+## Monitor the operator
+
+The Tempo Operator can expose upgrade and other operational metrics about the operator itself, and can create alerts based on these metrics.
+
+The operator can be configured via the ConfigMap `tempo-operator-manager-config` in the same namespace as the operator.
```
Comment (knylander-grafana) [inline docs/sources/tempo/setup/operator/monitor.md:104]:
```suggestion
The operator can be configured using the ConfigMap `tempo-operator-manager-config` in the same namespace as the operator.
```

**Review by knylander-grafana — APPROVED** [review]
Great additions! Thank you for updating. I've made a few suggestions, mostly incorporating some of the info from your replies. Good stuff. 

I'm approving for the documentation updates. However, there is a Go file that has some conflicts.

**Comment by andreasgerstmayr** [comment]
> Great additions! Thank you for updating. I've made a few suggestions, mostly incorporating some of the info from your replies. Good stuff.
> 
> I'm approving for the documentation updates.

Thank you so much for the review!

File: docs/sources/tempo/setup/operator/monitor.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/tempo/setup/operator/monitor.md]: Looks good! Thank you.

**Pull request merged**